src/nile/core/account.py

```diff
@@ -63,9 +65,59 @@ def deploy(self):
 
         return address, index
 
+    def declare(
+        self, contract_name, max_fee, nonce=None, alias=None, contracts_directory=None
+    ):
+        """Declare a contract through an Account contract."""
+        if contracts_directory is None:
+            contracts_directory = CONTRACTS_DIRECTORY
+
+        if nonce is None:
+            nonce = get_nonce(self.address, self.network)
+
+        if max_fee is None:
+            max_fee = 0
+        else:
+            max_fee = int(max_fee)
+
+        contract_class = compile_starknet_files(
+            files=[f"{contracts_directory}/{contract_name}.cairo"], debug_info=True
```

Comment:
do we need to compile in this location? It would be assumed that the contract being declared has been compiled before running this.